% !TeX spellcheck = en_US
% !TeX encoding = UTF-8
\documentclass{beamer}

\mode<presentation> { \usetheme{Madrid} }

\usepackage{graphicx, graphics}
\usepackage[notocbib]{apacite}
\usepackage[style=iso]{datetime2}
\usepackage{enumerate}
\DeclareGraphicsExtensions{.pdf, .png, .jpg, .gif}

\AtBeginSection[]
{
    \begin{frame}
        \vfill
        \centering
        \begin{beamercolorbox}[sep=8pt, center, shadow=true, rounded=true]{title}
            \usebeamerfont{title}
            \insertsectionhead
            \par
        \end{beamercolorbox}
        \vfill
    \end{frame}
}

\title[Periodontitis]{Periodontitis}

\author[Seunghoon Kim \and Jaewoong Lee]
{
    Seunghoon Kim
    \and
    Jaewoong Lee
    \and
    Semin Lee
}

\institute[UNIST]
{
    Ulsan National Institute of Science and Technology
    \medskip
    \newline
    \textit{jwlee230@unist.ac.kr}
}
\date{\today}

\begin{document}
    \begin{frame}
        \titlepage
    \end{frame}

    \begin{frame}
        \frametitle{Overview}
        \tableofcontents
    \end{frame}

    \section{Introduction}
    \begin{frame}
        \frametitle{Microbiome}

        \begin{itemize}
            \item Microbiota: the micro-organisms which live inside \& on humans \cite{microbiome1}
            \item Microbiome: about $10^{13}$ micro-organisms whose which collective genome \cite{microbiome2}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.3 \linewidth]{figures/microbiome.jpg}
            \caption{Concept of a core human microbiome \protect\cite{microbiome1}}
            \label{fig:microbiome}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{rRNA}

        \begin{itemize}
            \item Ribosomal RNA
            \item Well-known as a key to phylogeny \cite{rRNA1}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Periodontitis (Periodontal disease)}

        \begin{itemize}
            \item Clinical Attachment Loss \& Bone Loss \cite{periodontitis1}
            \item Risk Factors \cite{periodontitis2}
            \begin{enumerate}
                \item Smoking
                \item Diabetes
                \item Genetic factor
                \item Host response
            \end{enumerate}
        \end{itemize}
    \end{frame}

    \section{Materials}
    \begin{frame}
        \frametitle{16S rRNA Sequencing}

        \begin{itemize}
            \item 100 Healthy people
            \item 50 Chronic periodontitis -- Early
            \item 50 Chronic periodontitis -- Moderate
            \item 50 Chronic periodontitis -- Severe
        \end{itemize}
    \end{frame}

    \section{Methods}
    \begin{frame}
        \frametitle{QIIME2 Workflow}

        \begin{figure}
            \centering
            \includegraphics[width=0.7 \linewidth]{figures/qiime.png}
            \caption{QIIME2 Workflow \protect\cite{qiime1, qiime2}}
            \label{fig:qiime}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Denoising techniques}

        \begin{itemize}
            \item DADA2: Amplicon Sequence Variants (ASVs) \cite{DADA1}
            \item Deblur: Operational Taxonomic Units (OTUs) \cite{deblur1}
        \end{itemize}

        \begin{figure}
            \centering
            \includegraphics[width=0.4 \linewidth]{figures/denoising/denoising.pdf}
            \caption{Denoising Techniques}
            \label{fig:denoising}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Taxonomy Classification}

        \begin{itemize}
            \item Greengenes (GG) \cite{greengenes1}
            \item SILVA \cite{silva1}
            \item Human Oral Microbiome Database (HOMD) \cite{homd1}
        \end{itemize}

        \begin{figure}
            \centering
            \includegraphics[width=0.4 \linewidth]{figures/taxonomy/taxonomy.pdf}
            \caption{Taxonomy Classification}
            \label{fig:taxonomy}
        \end{figure}

        “A \textbf{higher} performance at taxonomic levels above \textit{genus} level; but performance appears to drop at \textit{species} level” \cite{performance1}
    \end{frame}

    \begin{frame}
        \frametitle{Merging Denosing and Taxonomy Classification}

        Merging multiple IDs (ASVs and OTUs) into one, which have:
        \begin{itemize}
            \item Different IDs.
            \item Identified as same taxonomy.
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/Merging/merging.pdf}
            \caption{Example Diagram for Merging Denosing and Taxonomy Classification}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Rarefaction}

        \begin{itemize}
            \item a statistical method of estimating the number of species expected in \textbf{a random sample} which taken from a collection \cite{rarefaction1}
            \item allows comparisons of \textbf{the species richness} among communities
            \item a good choice for \textbf{normalization} \cite{rarefaction2}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Alpha- \& Beta-diversity}

        \begin{itemize}
            \item Alpha-diversity: the richness of taxa \textbf{at a single community}
            \item Beta-diversity: the taxonomic differentiation \textbf{between communities}
        \end{itemize}

        \begin{figure}[p]
            \includegraphics[width=0.7 \linewidth]{figures/phylogenetic-info.jpg}
            \caption{Three Dimensions of Phylogenetic Information \protect\cite{phylogenetic1}}
        \end{figure}
   \end{frame}

    \begin{frame}
        \frametitle{Alpha-diversity}

        \begin{itemize}
            \item Evenness: a measurement of diversity in different type at community \cite{evenness1}
            \item Faith's Phylogenetic Diversity: a qualitative measurement of community richness which priorities for species conservation, incorporates with taxic diversity \cite{faith1}
            \item Observed Features: a number of observed taxa
            \item Shannon's diversity index: a significant aspect of community richness \cite{shannon1}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Beta-diversity}

        \begin{itemize}
            \item Bray-Curtis distance: a quantitative measurement of dissimilarity among communities \cite{bray1}
            \item Jaccard distance: a measurement of local distribution among communities \cite{jaccard1}
            \item Unweighted UniFrac distance: a qualitative measurement of phylogenetic distances \cite{unifrac1}
            \item Weighted UniFrac distance: a quantitative measurement of phylogenetic distances \cite{unifrac1}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{ANCOM}

        \begin{itemize}
            \item Analysis of composition of microbiomes
            \item ANCOM can be used for analyzing the composition of microbiomes in multiple populations \cite{ANCOM1}
            \item Differential abundance testing
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.3 \linewidth]{figures/ANCOM/example.png}
            \caption{Example ANCOM Volcano Plot \protect\cite{qiime1, qiime2}}
        \end{figure}

        \begin{itemize}
            \item clr: Centered log Ratio
            \item W: a count of the number of sub-hypothesis which have passed for given species
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Python Packages}

        \begin{itemize}
            \item Pandas \cite{pandas1}
            \item Scikit-learn \cite{sklearn1}
            \item Matplotlib \cite{matplotlib1, matplotlib2}
            \item Seaborn \cite{seaborn1}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{t-SNE}

        \begin{itemize}
            \item t-distributed stochastic neighbor embedding
            \item reveals high-dimensional data a location in two-dimensional map \cite{tSNE1}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.4 \linewidth]{figures/tSNE.png}
            \caption{Visualization by t-SNE \protect\cite{tSNE1}}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Classification}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/Classifier/classifier.pdf}
            \caption{Workflow of Classification}
        \end{figure}

        Classification Metrics:
        \begin{itemize}
            \item Accuracy
            \item Balanced Accuracy
            \item Sensitivity
            \item Specificity
            \item Precision
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/Classifier/best.pdf}
            \caption{Deciding the Best Features}
        \end{figure}
    \end{frame}

    \section{Results}
    \begin{frame}
        \frametitle{Quality Filter}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/Forward.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/Reverse.png}
                \\
                \mbox{(a) Forward Reads} & \mbox{(b) Reverse Reads}
            \end{array}$
            \caption{Sequence Quality Plot}
        \end{figure}

        $\therefore$ Maximum Sequence Length $n_{forword}$ = 300, $n_{reverse}$ = 265 \\
        $\because$ The longest length which has sequence quality $\ge 30$ at middle.
    \end{frame}

    \begin{frame}
        \frametitle{Rarefaction}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/Rarefaction/DADA.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/Rarefaction/Deblur.pdf}
                \\
                \mbox{(a) DADA2} & \mbox{(b) Deblur} \\
            \end{array}$
            \caption{Frequency per sample}
        \end{figure}

        $\therefore$ p-sampling-depth $n_{DADA2} = 3786$ and $n_{Deblur} = 7253$\\
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Alpha-diversity}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/DADA2/evenness.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/DADA2/faith.png}
                \\
                \mbox{(a) Evenness ($p < 0.01$)} & \mbox{(b) Faith PD ($p < 10^{-6}$)} \\

                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/DADA2/observed.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/DADA2/shannon.png}
                \\
                \mbox{(c) Observed features ($p < 10^{-3}$)} & \mbox{(d) Shannon's diversity ($p > 0.05$)}
            \end{array}$
            \caption{Alpha Diversity from DADA2 with Kruskal-Wallis among All Groups}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/Deblur/evenness.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/Deblur/faith.png}
                \\
                \mbox{(a) Evenness ($p < 0.05$)} & \mbox{(b) Faith PD ($p < 10^{-18}$)} \\

                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/Deblur/observed.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/AlphaDiversity/Deblur/shannon.png}
                \\
                \mbox{(c) Observed features ($p < 10^{-12}$)} & \mbox{(d) Shannon's diversity ($p < 10^{-4}$)}
            \end{array}$
            \caption{Alpha Diversity from Deblur with Kruskal-Wallis among All Groups}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Beta-diversity}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/DADA2.bray_curtis.png}
            \caption{Bray-Curtis Distance with DADA2}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/DADA2.jaccard.png}
            \caption{Jaccard Distance with DADA2}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/DADA2.unweighted_unifrac.png}
            \caption{Unweighted Unifrac Distance with DADA2}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/DADA2.weighted_unifrac.png}
            \caption{Weighted Unifrac Distance with DADA2}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/Deblur.bray_curtis.png}
            \caption{Bray-Curtis Distance with Deblur}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/Deblur.jaccard.png}
            \caption{Jaccard Distance with Deblur}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/Deblur.unweighted_unifrac.png}
            \caption{Unweighted Unifrac Distance with Deblur}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/BetaDiversity/Deblur.weighted_unifrac.png}
            \caption{Weighted Unifrac Distance with Deblur}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{ANCOM}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/ANCOM/DADA2.gg.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/ANCOM/DADA2.silva.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/ANCOM/DADA2.homd.png}
                \\
                \mbox{(a) GG (15 Taxa)} & \mbox{(b) SILVA (23 Taxa)} & \mbox{(c) HOMD (20 Taxa)} \\
            \end{array}$
            \caption{ANCOM Volcano Plot with DADA2}
        \end{figure}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/ANCOM/Deblur.gg.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/ANCOM/Deblur.silva.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/ANCOM/Deblur.homd.png}
                \\
                \mbox{(a) GG (27 Taxa)} & \mbox{(b) SILVA (20 Taxa)} & \mbox{(c) HOMD (28 Taxa)} \\
            \end{array}$
            \caption{ANCOM Volcano Plot with Deblur}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{t-SNE with Whole Microbiome}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/Whole/whole.DADA2.gg.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/Whole/whole.DADA2.silva.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/Whole/whole.DADA2.homd.png}
                \\
                \mbox{(a) GG (328 Taxa)} & \mbox{(b) SILVA (633 Taxa)} & \mbox{(c) HOMD (425 Taxa)} \\
            \end{array}$
            \caption{t-SNE Plot with Whole Microbiome from DADA2}
        \end{figure}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/Whole/whole.Deblur.gg.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/Whole/whole.Deblur.silva.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/Whole/whole.Deblur.homd.png}
                \\
                \mbox{(a) GG (232 Taxa)} & \mbox{(b) SILVA (414 Taxa)} & \mbox{(c) HOMD (235 Taxa)} \\
            \end{array}$
            \caption{t-SNE Plot with Whole Microbiome from Deblur}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{t-SNE with ANCOM Selected}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/ANCOM/ANCOM.DADA2.gg.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/ANCOM/ANCOM.DADA2.silva.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/ANCOM/ANCOM.DADA2.homd.png}
                \\
                \mbox{(a) GG (15 Taxa)} & \mbox{(b) SILVA (23 Taxa)} & \mbox{(c) HOMD (20 Taxa)} \\
            \end{array}$
            \caption{t-SNE Plot with ANCOM Selected from DADA2}
        \end{figure}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/ANCOM/ANCOM.Deblur.gg.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/ANCOM/ANCOM.Deblur.silva.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/ANCOM/ANCOM.Deblur.homd.png}
                \\
                \mbox{(a) GG (27 Taxa)} & \mbox{(b) SILVA (20 Taxa)} & \mbox{(c) HOMD (28 Taxa)} \\
            \end{array}$
            \caption{t-SNE Plot with ANCOM Selected from Deblur}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Random Forest Classifier -- Every Classes}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/whole.pdf}
            \caption{Random Forest Classifier Workflow}
        \end{figure}

        \textbf{Deblur + HOMD} gives the best result with balanced accuracy (0.778 with 13 taxa).

        \begin{table}
            \caption{Features Order by Importances}
            \begin{tabular}{c|l}
                Order & Taxonomy (Genus [Species]) \\ \hline
                0 & \textit{Porphyromonas gingivalis} \\
                1 & \textit{Actinomyces} \\
                2 & \textit{Pasteurellaceae} \\
                3 & \textit{Veillonella denticariosi} \\
                4 & \textit{Oribacterium sinus} \\
                5 & \textit{Peptostreptococcus anaerobius} \\
                6 & \textit{Prevotella nanceiensis} \\
                7 & \textit{Treponema} \\
                8 & \textit{Parvimonas sp. HMT 393} \\
                9 & \textit{Filifactor alocis} \\
                ... & ... \\
            \end{tabular}
        \end{table}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/ANCOM.Deblur.homd/metrics.png}
            \caption{Metrics by Feature Count}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/ANCOM.Deblur.homd/feature_0.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/ANCOM.Deblur.homd/feature_1.png}
                \\
                \mbox{(a) \textit{Porphyromonas gingivalis}} & \mbox{(b) \textit{Actinomyces}} \\
            \end{array}$
            \caption{Violin Plot by Features}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Random Forest Classifier -- Merge (Healthy+Early)}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/merge.pdf}
            \caption{Random Forest Classifier with Merging Workflow}
        \end{figure}

        \textbf{DADA2 + HOMD} gives the best result with balanced accuracy (0.938 with 11 taxa).

        \begin{table}
            \caption{Features Order by Importances}
            \begin{tabular}{c|l}
                Order & Taxonomy (Genus [Species]) \\ \hline
                0 & \textit{Porphyromonas gingivalis} \\
                1 & \textit{Actinomyces} \\
                2 & \textit{Prevotella intermedia} \\
                3 & \textit{Filifactor alocis} \\
                4 & \textit{Treponema sp. HMT 260} \\
                5 & \textit{Campylobacter showae} \\
                6 & \textit{Porphyromonas sp. HMT 285} \\
                7 & \textit{Peptostreptococcaceae XIG-6 nodatum} \\
                8 & \textit{Prevotella sp. HMT 304} \\
                9 & \textit{Actinomyces graevenitzii} \\
                10 & \textit{Tannerella forsythia} \\
            \end{tabular}
        \end{table}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/one.DADA2.homd/metrics.png}
            \caption{Metrics by Feature Count}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/one.DADA2.homd/feature_0.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/one.DADA2.homd/feature_1.png}
                \\
                \mbox{(a) \textit{Porphyromonas gingivalis}} & \mbox{(b) \textit{Actinomyces}} \\
            \end{array}$
            \caption{Violin Plot by Features}
        \end{figure}
   \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Random Forest Classifier -- Merge (Moderate + Severe)}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/merge.pdf}
            \caption{Random Forest Classifier with Merging Workflow}
        \end{figure}

        \textbf{Deblur + HOMD} gives the best result with balanced accuracy (0.777 with 28 taxa).

        \begin{table}
            \caption{Features Order by Importances}
            \begin{tabular}{c|l}
                Order & Taxonomy (Genus [Species]) \\ \hline
                0 & \textit{Porphyromonas gingivalis} \\
                1 & \textit{Actinomyces} \\
                2 & \textit{Pasteurellaceae} \\
                3 & \textit{Veillonella denticariosi} \\
                4 & \textit{Oribacterium sinus} \\
                5 & \textit{Peptostreptococcus anaerobius} \\
                6 & \textit{Treponema} \\
                7 & \textit{Prevotella nanceiensis} \\
                8 & \textit{Porphyromonas sp. HMT 285} \\
                9 & \textit{Parvimonas sp. HMT 393} \\
                ... & ... \\
            \end{tabular}
        \end{table}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/two.Deblur.homd/metrics.png}
            \caption{Metrics by Feature Count}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/two.Deblur.homd/feature_0.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/two.Deblur.homd/feature_1.png}
                \\
                \mbox{(a) \textit{Porphyromonas gingivalis}} & \mbox{(b) \textit{Actinomyces}} \\
            \end{array}$
            \caption{Violin Plot by Features}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Random Forest Classifier -- Merge (H + E) \& (M + S)}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/merge.pdf}
            \caption{Random Forest Classifier with Merging Workflow}
        \end{figure}

        \textbf{DADA2 + SILVA} gives the best result with balanced accuracy (0.934 with 15 taxa).

        \begin{table}
            \caption{Features Order by Importances}
            \begin{tabular}{c|l}
                Order & Taxonomy (Genus [Species]) \\ \hline
                0 & \textit{Actinomyces} \\
                1 & \textit{Treponema denticola} \\
                2 & \textit{Porphyromonas gingivalis} \\
                3 & \textit{Prevotella intermedia} \\
                4 & \textit{Filifactor alocis} \\
                5 & \textit{Schaalia odontolytica} \\
                6 & \textit{Tannerella forsythia} \\
                7 & \textit{Eubacterium nodatum} \\
                8 & \textit{Corynebacterium durum} \\
                9 & \textit{Oribacterium} \\
                ... & ... \\
            \end{tabular}
        \end{table}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/one-two.DADA2.silva/metrics.png}
            \caption{Metrics by Feature Count}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/one-two.DADA2.silva/feature_0.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/RandomForest/one-two.DADA2.silva/feature_1.png}
                \\
                \mbox{(a) \textit{Actinomyces}} & \mbox{(b) \textit{Treponema denticola}} \\
            \end{array}$
            \caption{Violin Plot by Features}
        \end{figure}
    \end{frame}

    \section{Discussion}
    \begin{frame}
        \frametitle{\textit{Actinomyces} Genus}
    \end{frame}

    \begin{frame}
        \frametitle{\textit{Porphyromonas} Genus}
    \end{frame}

    \begin{frame}
        \frametitle{\textit{Treponema} Genus}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{References}
        \bibliographystyle{apacite}
        \bibliography{reference}
    \end{frame}
\end{document}